dipy/tracking/utils.py

```diff
@@ -595,6 +596,56 @@ def target(streamlines, target_mask, affine, include=True):
             yield sl
 
 
+@_with_initialize
+def target_line_based(streamlines, target_mask, affine=None, include=True):
+    """Filters streamlines based on whether or not they pass through a ROI,
+    using a line-based algorithm. Mostly used as a remplacement of `target`
+    for compressed streamlines.
+
+    This function never returns single-point streamlines, wathever the
+    value of `include`.
+
+    Parameters
+    ----------
+    streamlines : iterable
+        A sequence of streamlines. Each streamline should be a (N, 3) array,
+        where N is the length of the streamline.
+    target_mask : array-like
+        A mask used as a target. Non-zero values are considered to be within
+        the target region.
+    affine : array (4, 4)
+        The affine transform from voxel indices to streamline points.
+    include : bool, default True
+        If True, streamlines passing through `target_mask` are kept. If False,
+        the streamlines not passing through `target_mask` are kept.
+
+    Returns
+    -------
+    streamlines : generator
+        A sequence of streamlines that pass through `target_mask`.
+
+    References
+    ----------
+    [Bresenham5] Bresenham, Jack Elton. "Algorithm for computer control of a
+                 digital plotter", IBM Systems Journal, vol 4, no. 1, 1965.
+    [Houde15] Houde et al. How to avoid biased streamlines-based metrics for
+              streamlines with variable step sizes, ISMRM 2015.
+
+    See Also
+    --------
+    density_map
```

Comment:
Hi @nilgoyyou and @jchoude , great job! 
And apologies for delaying to review this PR. Bad timing. 
Can I ask to add the compression function to the see also? You may want to use the full module path when writing in see also to make it easier for the user to know where to find things. For example I would write
dipy.tracking.utils.density_map
dipy.tracking.streamline.compress_streamlines